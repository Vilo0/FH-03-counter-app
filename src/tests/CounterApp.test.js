import '@testing-library/react';
import { shallow } from 'enzyme/build';
import CounterApp from '../CounterApp';

describe('Pruebas en <CounterApp />', () => {

    test('debe de mostrar <CounterApp /> correctamente y su valor por defecto', () => {
      
        const initial = 5;
        const wrapper = shallow( <CounterApp /> );

        expect(wrapper).toMatchSnapshot();

        const contador = parseInt(wrapper.find('h2').text());
        expect(contador).toBe(initial);

    })

    test('debe de mostrar <CounterApp /> correctamente y el valor por defecto 100', () => {
      
        const wrapper = shallow( <CounterApp value={100} /> );

        const contador = wrapper.find('h2').text().trim();
        expect(contador).toBe('100');

    });

});