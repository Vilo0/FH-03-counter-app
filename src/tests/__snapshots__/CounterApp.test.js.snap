// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente y el valor por defecto 100 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;

exports[`Pruebas en <CounterApp /> debe de mostrar <CounterApp /> correctamente y su valor por defecto 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    5
  </h2>
  <button
    onClick={[Function]}
  >
    -1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    +1
  </button>
</Fragment>
`;
